invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/UserDetails/components/ItemsSearch/ItemsSearch.js

```diff
@@ -32,28 +34,24 @@ export default class ItemsSearch extends Component {
     }
   }
 
-  onInputChange = queryString => {
-    this.updateQueryString(queryString);
-  };
-
   executeSearch = queryString => {
+    queryString = queryString || this.state.currentQueryString;
+    this.updateQueryString(queryString);
     if (queryString) {
       return this.fetchItems(queryString);
-    } else {
-      return this.fetchItems(this.props.queryString);
     }
   };
 
   _onPasteHandler = e => {
+    const hits = _get('this.props.items.hits');
     let queryString = e.clipboardData.getData('Text');
+    this.setState({ currentQueryString: '' });
     this.executeSearch(queryString).then(data => {
-      if (this.props.items.hits[0].status === 'LOANABLE') {
-        this.checkoutItem(this.props.items.hits[0], this.props.patron).then(
-          () => {
-            this.clearResults();
-            this.fetchUpdatedCurrentLoans(this.props.patron);
-          }
-        );
+      if (hits && hits[0].status === 'LOANABLE') {
```

Comment:
Shouldn't we also check if hits[0] is truthy? Otherwise .status will be called on undefined.

Reply:
Yes, you're right. Thanks for spotting that!